src/plugins/discover/public/application/components/top_nav/open_options_popover.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { I18nStart } from 'kibana/public';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiSpacer, EuiButton, EuiText, EuiWrappingPopover, EuiCode } from '@elastic/eui';
+import { getServices } from '../../../kibana_services';
+import './open_options_popover.scss';
+
+let isOpen = false;
+
+interface OptionsPopoverProps {
+  onClose: () => void;
+  anchorElement: HTMLElement;
+}
+
+export function OptionsPopover(props: OptionsPopoverProps) {
+  const {
+    core: { uiSettings },
+    addBasePath,
+  } = getServices();
+  const isLegacy = uiSettings.get('doc_table:legacy');
+
+  const mode = isLegacy
+    ? i18n.translate('discover.openOptionsPopover.dataGridText', {
+        defaultMessage: 'Data grid',
+      })
+    : i18n.translate('discover.openOptionsPopover.legacyTableText', {
+        defaultMessage: 'Legacy table',
+      });
```

Comment:
this should be inverted to like

isLegacy ? 'Legacy table' : 'Data grid'

Reply:
You are so right Matthias 😱

Reply:
Done :D